### ISSUE TITLE
.NET Core 3 support

### ISSUE THREAD

**Issue by TFTomSun** [filipw/dotnet-script#473]
Can you please add .net core 3 support for the global tool? Background: I use scripting in a CI/CD pipeline. Since I upgraded to the .net core 3 docker image from microsoft, i can't install the dotnet script tool anymore. I think its just a small change in the TargetFrameworks msbuild property to support both, .net core 2.1 and .net core 3.

**Comment by seesharper** [comment]
We are currently working with .Net Core 3 support.
It is however not as easy as adding another target framework.
The 3.0 SDK introduces some changes with regards to how we collect runtime assemblies for executing a script.  This also affects compilation assemblies which we need in OmniSharp to provide intellisense and language services in OmniSharp (C# extension).

You can follow this branch for the progress on this.

https://github.com/filipw/dotnet-script/tree/feature/netcoreapp3.0

**Comment by maxhamulyak** [comment]
Is there any way to run a preview version of the .Net Core 3 version?

**Comment by robertmclaws** [comment]
Looking forward to this! I have built an EF 6.3 code generator for .NET Core using CSX files, but the assembly loading mechanism in 3.0 is causing some problems. You can see it demonstrated here: https://github.com/robertmclaws/EDMXParsingSample.

I'm actually glad it happened because it exposed a flaw in the EF6.3 metadata parsing code, but it's slowed my plans for releasing my framework. I'm generating the code with a `dotnet tool` in the meantime, but will be great when the scripts can work too.

Thanks for all your efforts on this project!

**Comment by TheSimpleZ** [comment]
> We are currently working with .Net Core 3 support.
> It is however not as easy as adding another target framework.
> The 3.0 SDK introduces some changes with regards to how we collect runtime assemblies for executing a script. This also affects compilation assemblies which we need in OmniSharp to provide intellisense and language services in OmniSharp (C# extension).
> 
> You can follow this branch for the progress on this.
> 
> https://github.com/filipw/dotnet-script/tree/feature/netcoreapp3.0

The branch you're mentioning in your link is gone. What is the progress so far?

**Comment by seesharper** [comment]
We have just released version 0.50.0 adding support for .Net Core 3.0

**Comment by KyleMit** [comment]
> We have just released version 0.50.0 adding support for .Net Core 3.0

@seesharper , are there any upgrade instructions?

From what I can tell, the two places that need updates are in the omnisharp.json and launch.json file

#### `omnisharp.json`

```diff
{
  "script": {
    "enableScriptNuGetReferences": true,
-   "defaultTargetFramework": "netcoreapp3.1"
+   "defaultTargetFramework": "netcoreapp3.1"
  }
}
```

#### `launch.json`

```diff
{
    "name": ".NET Script",
    "type": "coreclr",
    "request": "launch",
    "program": "dotnet",
    "args": [
        "exec",
-       "~/.dotnet/tools/.store/dotnet-script/0.52.0/dotnet-script/0.52.0/tools/netcoreapp2.1/any/dotnet-script.dll",
+       "~/.dotnet/tools/.store/dotnet-script/0.52.0/dotnet-script/0.52.0/tools/netcoreapp3.1/any/dotnet-script.dll",
        "${file}"
    ],
    "cwd": "${workspaceRoot}",
},
```

**Comment by seesharper** [comment]
`dotnet script init` in the script folder should update those values.. If not try to delete the `omnisharp.json` file and the `.vscode` folder and then do a `dotnet script init`
